### PR TITLE
[WIP] Behave in docker introduced

### PR DESCRIPTION
**Pull request by magul**
Based on:
 * https://eshlox.net/2016/11/22/dockerize-behave-selenium-tests/
 * https://sourabhbajaj.com/blog/2017/02/07/gui-applications-docker-mac/

### CODE REVIEW

File: behave/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,13 @@
+FROM python:3.7
+
+RUN wget -q -O - https://dl-ssl.google.com/linux/linux_signing_key.pub | apt-key add -
+RUN sh -c 'echo "deb [arch=amd64] http://dl.google.com/linux/chrome/deb/ stable main" >> /etc/apt/sources.list.d/google.list'
+RUN apt-get update -y && \
+    apt-get install -y google-chrome-stable chromedriver
```
Comment (w1stler) [inline behave/Dockerfile]: You might want to freeze `chromedriver` (as well as selenium and behave below).

File: docker-compose.yml [hunk 2]
```diff
@@ -48,5 +48,14 @@ services:
     depends_on:
       - backend
 
+  behave:
+    build: behave
+    shm_size: 256m
```
[the remark this answers is not in the document]
Comment (magul) [reply docker-compose.yml]: As here: https://eshlox.net/2016/11/22/dockerize-behave-selenium-tests/
```
We need shm_size to set size of /dev/shm and DBUS_SESSION_BUS_ADDRESS set to /dev/null, without that I had a lot of issues with running Chrome. 
```

I just assumed that it should be set. We can try and remove it.

File: Makefile [hunk 3]
```diff
@@ -41,6 +41,11 @@ test-backend:
 test:
 	make test-backend
 
+test-behave:
+	xhost +si:localuser:root
+	docker-compose exec behave behave --tags=-skip
```
[the remark this answers is not in the document]
Comment (magul) [reply Makefile]: here actually no part of this command is directory name. First `behave` is a name of docker container in `docker-compose.yml` and the second is the name of a command. By default `behave` use `features` directory and from context of docker container it is using it and there are `feature` files.

File: behave/Dockerfile (hunk 1 above)
Comment (magul) [reply behave/Dockerfile]: I would do that if I would be able to install `chromedriver` as pip package. Don't actually know how to do that (and if that's possible) from a perspective od debian system-based package.

Comment (w1stler) [reply behave/Dockerfile]: You can use https://chromedriver.storage.googleapis.com/, there are all of the versions.

File: docker-compose.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (w1stler) [reply docker-compose.yml]: 👍

File: Makefile (hunk 3 above)
[the remark this answers is not in the document]
Comment (w1stler) [reply Makefile]: We have `features` file in the main folder of repository. I mean that should we create directory related to behave tests, so we would have (maybe even git submodule).
```
- frontend
- backend
- featuers <--- this shouldn't be here
```
Look at the structure eg. here: https://behave.readthedocs.io/en/latest/tutorial.html#features (separate folders for features, files, etc. to keep it clean).

**Comment by adamgajzlerowicz** [comment]
Are we getting this merged, or does it need more work?

**Comment by magul** [comment]
@adamgajzlerowicz it needs a little bit more work. I put `[WIP]` at the beginning of title here.

**Comment by w1stler** [comment]
Not applicable at the moment.

**Pull request closed without merging**